nbextensions/bulkImportCell/util.js

```diff
@@ -53,24 +53,88 @@ define(['common/runtime', 'StagingServiceClient'], (Runtime, StagingServiceClien
+     * populated with options for file inputs.
+     *
+     * For example, if the filePath parameters in the data model looks like this:
+     * [
+     *   { input_file: 'file1', output_obj: 'obj1' },
+     *   { input_file: 'file2', output_obj: 'obj2' }
+     * ]
+     *
+     * where the fileParamIds are ['input_file', 'output_obj'], and the outputParamIds
+     * Array is just ['output_obj'], and we have a few missing files in a Set, {'file1', 'file3'},
+     * this would return:
+     * [
+     *   {
+     *      input_file: {
+     *          invalidValues: { 'file1', 'file3' }
+     *      }
+     *   },
+     *   {
+     *      input_file: {
+     *          invalidValues: { 'file1', 'file3' }
+     *      }
+     *   }
+     * ]
+     *
+     * This still seems a little bit arbitrary, but as long as the file inputs are validated as
+     * text (and not data objects), this will continue to work.
+     * @param {Object} model the main bulk import cell data model
+     * @param {string} fileType which filetype to build options for
+     * @param {Set} missingFiles the set of files that were present when this bulk import cell
+     *   was created, but are now missing from the server
+     * @returns {Array[Object]} each object has a key for each file input (not output). Values
+     *   are the validation options for each mapped input
+     */
+    function getFilePathOptionsForValidation(model, fileType, missingFiles) {
+        let fpIds = model.getItem(['app', 'fileParamIds', fileType]);
+        const outIds = model.getItem(['app', 'outputParamIds', fileType]);
+        // overwrite fpIds with JUST the file inputs (not the outputs)
+        fpIds = new Set(fpIds.filter((id) => !outIds.includes(id)));
+        const fpVals = model.getItem(['params', fileType, 'filePaths']);
+
+        // fpVals = Array of input file path rows from the importer for the current fileType
+
+        return fpVals.map((filePath) => {
+            const fpOptions = {};
+            for (const id of Object.keys(filePath)) {
+                fpOptions[id] = {};
+                if (fpIds.has(id)) {
+                    fpOptions[id].invalidValues = missingFiles;
+                }
+            }
+            return fpOptions;
+        });
+    }
+
     /**
      * Runs evaluateAppConfig over all apps, given the BulkImportCell's model, and constructs
      * the ready state.
+     * @param {Object} model the data model from the BulkImportCell, containing all the cell info
+     * @param {Object} specs keys = app id, values = app specs
+     * @param {Set} missingFiles Set of files that the app cell expects to be present,
+     *  but are missing from the staging area
+     * @returns
      */
-    function evaluateConfigReadyState(model, specs) {
+    function evaluateConfigReadyState(model, specs, missingFiles) {
+        // given some missing files (precalcuate for now), set up options for evaluating all file inputs
```

Comment:
Precalculate "for now" = it would be nice to have that optional. If left as `undefined`, we should look that up from a cache, or just make the call based on the list of input files in the `model` object.

However, it's only actually used in two places (the `configure` and `bulkImportCell` modules), and it seemed cleaner to just make both of those places look up their own missing files for now. The `Configure` module needs to keep a hold of that list of missing files to compare while the user is setting up inputs.